### PR TITLE
SALTO-4177 Support deploy of issue layout

### PR DESCRIPTION
**Pull request by idoamit8**
This PR supports issueLayout deploy via a private API and small fetch fixes. 

---
It follows this PR for fetching issueLayouts -  https://github.com/salto-io/salto/pull/4709

---
_Release Notes_: 
Now support deploy of issue Layout. 

---
_User Notifications_: 
Jira:
Upon your next fetch, you may encounter new instances of type **IssueLayout**.

### CODE REVIEW

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts [hunk 1]
```diff
@@ -113,6 +120,44 @@ const getProjectToScreenMapping = async (elements: Element[]): Promise<Record<st
...
+    return undefined
+  }).filter(isDefined)
+
+  if (isResolvedReferenceExpression(issueLayout.value.projectId)
+  && isResolvedReferenceExpression(issueLayout.value.extraDefinerId)) {
+    const data = {
+      projectId: issueLayout.value.projectId.value.value.id,
+      extraDefinerId: issueLayout.value.extraDefinerId.value.value.id,
+      issueLayoutType: 'ISSUE_VIEW',
+      owners: [],
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:150]: Aren't you supposed to pass here the issue types the layout is relevet to?

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts [hunk 2]
```diff
@@ -113,6 +120,44 @@ const getProjectToScreenMapping = async (elements: Element[]): Promise<Record<st
...
+      issueLayoutConfig: {
+        items,
+      },
+    }
+    const url = `/rest/internal/1.0/issueLayouts/${issueLayout.value.id}`
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:155]: What if the issue layout is an addition? (and won't have an id)

**Review by alonstern — COMMENTED** [review]
👍

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts (hunk 1 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:150]: added `issueLayoutType.fields.owners.annotations[CORE_ANNOTATIONS.UPDATABLE] = false`

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts [hunk 3]
```diff
@@ -113,20 +115,64 @@ const getProjectToScreenMapping = async (elements: Element[]): Promise<Record<st
...
+        client })
+      if (!isIssueLayoutResponse(response.data)) {
+        return undefined
+      }
+      issueLayout.value.id = response.data.issueLayoutConfiguration.issueLayoutResult.id
+    }
+    const url = `/rest/internal/1.0/issueLayouts/${issueLayout.value.id}`
+    await client.put({ url, data })
+  }
+  return undefined
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:162]: It means if the projectId or extraDefinerId are not references you are going to do nothing but tell the user the change was deployed. You probably should return some Error to the user here

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts [hunk 4]
```diff
@@ -113,20 +115,64 @@ const getProjectToScreenMapping = async (elements: Element[]): Promise<Record<st
...
+      issueLayoutConfig: {
+        items,
+      },
+    }
+    if (isAdditionChange(change)) {
+      const response = await getIssueLayout({ projectId: issueLayout.value.projectId.value.value.id,
+        screenId: issueLayout.value.extraDefinerId.value.value.id,
+        client })
+      if (!isIssueLayoutResponse(response.data)) {
+        return undefined
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:155]: You should return an error to the user in this case

**Review by alonstern — COMMENTED** [review]
Last comments 👌

**Review by alonstern — APPROVED** [review]
🔥 
Before merging, is there a noise reduction for this

**Comment by tamtamirr** [comment]
@idoamit8 The **User Notifications** section should describe expected changes to existing workspaces.
I would suggest something like:
```
Jira:
Upon your next fetch, you may encounter new instances of type **IssueLayout**.
```

Moreover, if this is going to be "noisy" we probably want to consider a **PROD** ticket here to communicate this change to customers, for **git audit** use-case.

**Pull request merged**